Tests/DatadogTests/Datadog/RUM/RUMMonitor/Scopes/RUMResourceScopeTests.swift

```diff
@@ -51,6 +51,9 @@ class RUMResourceScopeTests: XCTestCase {
         )
 
         currentTime.addTimeInterval(2)
+        let expectedActionID = parent.context.activeUserActionID
+        parent.context.activeUserActionID = .mockRandom()
+        parent.context.activeViewID = .mockRandom()
```

Comment:
It's hard to understand as this test will pass normally without this code. It only replaces the action ID so we prove that this has no impact on the `model.action.id`. IMO this should be a separate test case explaining what we want to test using clear given / when / then.

Reply:
the description of `ResourceScope` seems to be
1. keep the initial active action
2. get the final active view
this case verifies the description of component

i can remove if it's confusing.?

Reply:
As we add the new behaviour: _**given** the Resource started on User Action, **when** new Action is started, **it** tracks the action which caused it_, we should have the unit test for it.

I just mean that we test it in a wrong way: by modifying the existing test which asserts totally different behaviour. Removing this code will still make this test pass.

Reply:
Well, the definition of ResourceScope should be that it should always be linked with the initial active context (both view and action). If another view is started, it should definitely not be linked with the resource.

Reply:
as `ResourceScope` takes `context` in `init` now, we don't need to test context mutation during life-cycle anymore: problem solved automatically ✅